lib/LaTeXML/Core/Document.pm

```diff
@@ -1443,11 +1443,32 @@ sub pruneXMDuals {
       $self->compactXMDual($dual, $content, $presentation); } }
   return; }
 
+our @CONTENT_TRANSFER_ATTRS = qw(decl_id meaning name omcd);
+
 sub compactXMDual {
   my ($self, $dual, $content, $presentation) = @_;
-  # For now only handle compacting mirror applications
-  return if ($self->getNodeQName($content) ne 'ltx:XMApp') ||
-    ($self->getNodeQName($presentation) ne 'ltx:XMApp');
+  my $c_name = $self->getNodeQName($content);
+  my $p_name = $self->getNodeQName($presentation);
+  # 1.Quick fix: merge two tokens
+  if (($c_name eq 'ltx:XMTok') && ($p_name eq 'ltx:XMTok')) {
+    for my $attr_key (@CONTENT_TRANSFER_ATTRS) {
+      if (my $attr_val = $content->getAttribute($attr_key)) {
+        $content->removeAttribute($attr_key);
+        $presentation->setAttribute($attr_key, $attr_val); } }
+    # if the dual has any attributes migrate them to the new XMTok
+    my %transfer_attrs = ();
+    for my $attr ($dual->attributes) {
+      if ($attr->nodeType == XML_ATTRIBUTE_NODE) {
+        $transfer_attrs{ $attr->nodeName } = $attr->getValue; } }
+    $self->replaceNode($dual, $presentation);
+    # transfer the attributes after replacing, so that the bookkeeping has been undone
+    for my $key (keys %transfer_attrs) {
+      if (($key eq 'xml:id') || !$presentation->getAttribute($key)) {
```

Comment:
Some of the dual's attributes should override the presentation's attributes, certainly `role`, `meaning`, probably anything from `@CONTENT_TRANSFER_ATTRS`.

Also, theoretically if more than 1 of (dual,presentation,content)have ids, all referrers to the extras should get updated to use the one you keep.

Reply:
Thanks, hadn't thought that part through to be honest, and the changes to the tests are Good - `\Langle` from mathbbol and `\Ydown` from stmaryrd now once again have the correct roles.

Reply:
Ready for re-review